libs/ui-image-upload-dialog/src/lib/ui-image-upload-dialog/ui-image-upload-dialog.component.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  inject,
+  signal,
+  ViewChild,
+} from '@angular/core';
+import { MatButtonModule } from '@angular/material/button';
+import { MAT_DIALOG_DATA, MatDialogModule } from '@angular/material/dialog';
+import { MatIconModule } from '@angular/material/icon';
+import { RecipeDto, RecipesUiActions } from '@hoa-recipes/data-access-recipes';
+import { Store } from '@ngrx/store';
+
+@Component({
+  selector: 'hoa-recipes-ui-image-upload-dialog',
+  standalone: true,
+  imports: [MatDialogModule, MatButtonModule, MatIconModule],
+  templateUrl: './ui-image-upload-dialog.component.html',
+  styleUrl: './ui-image-upload-dialog.component.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class UiImageUploadDialogComponent {
+  private store = inject(Store);
+  dialogData: RecipeDto = inject(MAT_DIALOG_DATA);
+
+  @ViewChild('fileInput') fileInputRef!: ElementRef<HTMLInputElement>;
```

Comment:
use [viewChild API](https://angular.io/api/core/viewChild)

Reply:
then `uploadButtonDisabled` would become a `computed` property and would also remove methods like  `updateUploadButtonStatus`